app/assets/javascripts/components/CreateCase.jsx

```diff
@@ -0,0 +1,213 @@
+class CreateCase extends React.Component {
+    constructor(props) {
+        super();
+        this._handleChange = this._handleChange.bind(this);
+        this._handleSubmit = this._handleSubmit.bind(this);
+        this.state = {
+            showForm: true,
+            message: '',
+            error: ''
```

Comment:
you need to initialize all the states you populate later in the file here, but if you follow Ken's suggestion you can add a `case` state object here instead of having to do each input field separately